pandas/_libs/groupby_helper.pxi.in

```diff
@@ -444,6 +444,39 @@ def group_nth_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
                 else:
                     out[i, j] = resx[i, j]
 
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def group_rank_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
+                        ndarray[{{c_type}}, ndim=2] values,
+                        ndarray[int64_t] labels,
+                        bint is_datetimelike, **kwargs):
+    """
```

Comment:
looks ok. don't pass kwargs generally around (esp in cython). we want to specify the args directly. this may make the call slightly more tricky in cython_transform (e.g. you may need to use a partial or lambda to hold the additional args)